guide/additional-info/changes-in-v13.md

```diff
@@ -0,0 +1,1095 @@
+
+### Bitfields / Permissions
+
+Bitfields are now `BigInt`s instead of `Number`s. This can be handled using the `BigInt()` class, or the n-suffixed BigInt literal.
+
+```diff
+- const p = new Permissions(104324673);
++ const p = new Permissions(BigInt(104324673)); // using class
++ const p = new Permissions(104324673n); // using literal
+```
+In additional, the usage of string literals for bitfield flags such as `Permissions` and `UserFlags` is discouraged; you should use the flag instead.
+
+```diff
+- permissions.has('SEND_MESSAGES')
++ permissions.has(Permissions.FLAGS.SEND_MESSAGES)
+```
+
+### DM Channels
+
+On Discord API v8, DM Channels do not emit the `CHANNEL_CREATE` event, which means Discord.js is unable to cache them automatically. In order for your bot to receive DMs the `CHANNEL` partial must be enabled.
+
+### Webpack
+
+Webpack builds are no longer supported.
+
+## Changes and deletions
+
+### APIMessage
+
+The APIMessage class has been renamed to `MessagePayload`, resolving a naming clash with an interface in the `discord-api-types` library which raw message data objects.
+
+### Client
+
+#### Client#emojis
+
+The Client Emoji manager is now a `BaseGuildEmojiManager`, providing cache resolution only and removing methods that would fail to create emojis as there was no Guild context.
+
+#### Client#fetchApplication
+
+The `Client#fetchApplication` has been removed and replaced with the `Client#application` property.
+
+```diff
+- client.fetchApplication().then(application => console.log(application.name))
++ console.log(client.application.name);
+```
+
+#### Client#generateInvite
+
+`Client#generateInvite` no longer supports `PermissionsResolvable` as its argument, requiring `InviteGenerationOptions`.
+
+`InviteGenerationOptions` requires that at least one of either `bot` or `applications.commands` is provided in `scopes` to generate a valid invite URL.
+
+To generate an invite link for Slash Commands only.
+
+```js
+client.generateInvite({ scopes: ['applications.commands'] });
+```
+
+To generate an invite link for a bot and define required permissions.
+
+```diff
+- client.generateInvite([Permissions.FLAGS.SEND_MESSAGES]);
++ client.generateInvite({ scopes: ['bot'], permissions: [Permissions.FLAGS.SEND_MESSAGES] })
+```
+
+### Client#login
+
+Previously when a token had reached its 1000 login limit for the day, discord.js would treat this as a rate limit and silently wait to login again, but this was not communicated to the user.
+
+This will now instead cause an error to be thrown.
+
+### ClientOptions
+
+#### ClientOptions#fetchAllMembers
+
+The `ClientOptions#fetchAllMembers` option has been removed.
+
+With the introduction of gateway intents, the `fetchAllMembers` Client option would often fail and causes significant delays in ready states or even cause timeout errors.
```

Comment:
```suggestion
With the introduction of gateway intents, the `fetchAllMembers` Client option would often fail and cause significant delays in ready states or even cause timeout errors.
```
Not a 100% sure about this one, but it sounded weird

Reply:
no, semantical difference. your suggestion implies that the significant delay is only happening when the fail does.
initial means that it would often fail and (generally) causes significant delays